src/main/java/org/springframework/data/redis/support/collections/DefaultRedisMap.java

```diff
@@ -155,20 +156,7 @@ public boolean containsValue(Object value) {
 	 */
 	@Override
 	public Set<java.util.Map.Entry<K, V>> entrySet() {
-
-		Set<K> keySet = keySet();
-		checkResult(keySet);
-		Collection<V> multiGet = hashOps.multiGet(keySet);
-
-		Iterator<K> keys = keySet.iterator();
-		Iterator<V> values = multiGet.iterator();
-
-		Set<Map.Entry<K, V>> entries = new LinkedHashSet<>();
-		while (keys.hasNext()) {
-			entries.add(new DefaultRedisMapEntry(keys.next(), values.next()));
-		}
-
-		return entries;
+		return hashOps.entries().entrySet();
```

Comment:
Nice optimization.

Reply:
Thanks. In the first place, I didn't intend this as an optimization, but as a fix to be able to fetch more than 1024*1024 -1 entries. But I agree, the optimization is a nice side effect ;)
I re-added the `checkResult(...)`.